resources/event-sources/templates/100-pre-upgrade-job.yaml

```diff
@@ -0,0 +1,136 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+rules:
+  - apiGroups: ["extensions"]
+    resources: ["deployments"]
+    verbs: ["get", "delete"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get", "delete"]
+  - apiGroups: [""]
+    resources: ["serviceaccounts"]
+    verbs: ["get", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Do we still need this `Rolebinding`?

Reply:
We have two roles defined "event-sources-pre-upgrade" and "event-sources-pre-upgrade-helm" and only the last one is a ClusterRole. This is why it is more restricted. Both roles are bound the the same service account: "event-sources-pre-upgrade"

Reply:
Changed as proposed, using only Helm now.